pontoon/sync/vcs/repositories.py

```diff
@@ -16,44 +16,55 @@ class PullFromRepositoryException(Exception):
 
 class PullFromRepository(object):
 
-    def __init__(self, source, target):
+    def __init__(self, source, target, branch):
         self.source = source
         self.target = target
+        self.branch = branch
 
     def pull(self, source=None, target=None):
         raise NotImplementedError
 
 
 class PullFromGit(PullFromRepository):
 
-    def pull(self, source=None, target=None):
+    def pull(self, source=None, target=None, branch=None):
         log.debug("Git: Update repository.")
 
         source = source or self.source
         target = target or self.target
+        branch = branch or self.branch
 
         command = ["git", "fetch", "--all"]
         execute(command, target)
 
+        remote = "origin"
+        if branch:
+            remote += "/" + branch
```

Comment:
I was on the fence about this one, actually. So glad to have a decision made on this ;)